pkg/framework/aries/default.go

```diff
@@ -49,52 +38,43 @@ func (d *defFramework) didResolverProvider() (DIDResolver, error) {
 	return resl, nil
 }
 
-func (d *defFramework) storeProvider() (storage.Provider, error) {
-	if d.storeProv != nil {
-		return d.storeProv, nil
-	}
-
-	// TODO - Need to configure the path externally
-	storeProv, err := leveldb.NewProvider(DBPath)
+func storeProvider() (storage.Provider, error) {
+	storeProv, err := leveldb.NewProvider(dbPath)
 	if err != nil {
 		return nil, fmt.Errorf("leveldb provider initialization failed : %w", err)
 	}
-
-	d.storeProv = storeProv
 	return storeProv, nil
 }
 
 // defFrameworkOpts provides default framework options
-func defFrameworkOpts() ([]Option, error) {
-	// get the default framework configs
-	def := defFramework{}
+func defFrameworkOpts(frameworkOpts *Aries) error {
 
-	var opts []Option
 	// protocol provider factory
-	opt := WithTransportProviderFactory(def.transportProviderFactory())
-	opts = append(opts, opt)
-
-	reslv, err := def.didResolverProvider()
-	if err != nil {
-		return nil, fmt.Errorf("resolver initialization failed : %w", err)
+	if frameworkOpts.transport == nil {
+		frameworkOpts.transport = transportProviderFactory()
 	}
-	opts = append(opts, WithDIDResolver(reslv))
-
-	storeProv, err := def.storeProvider()
+	if frameworkOpts.storeProvider == nil {
+		storeProv, err := storeProvider()
```

Comment:
I assume there is a TODO to move these default providers to the sub-package?

Reply:
#209